Classes/KIFUITestActor.m

```
@@ -676,18 +683,126 @@ - (void)selectPickerViewRowWithTitle:(NSString *)title inComponent:(NSInteger)co
+        NSString *dayOfWeekMonthDay = datePickerColumnValues[0];
+        // in a date time picker the first value is either "Today" or something like "Tue Aug 12" so seperate by a space to
+        // grab the correct values from it.
+        NSArray<NSString *>*dayOfWeekMonthDayComponents = [dayOfWeekMonthDay componentsSeparatedByString:@" "];
+        NSDateComponents *currentDateComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:datePicker.date];
+        // if the string components of the first value is 1 we assume it's saying "Today" and use the current dates components
+        if(dayOfWeekMonthDayComponents.count == 1) {
+            NSDateComponents *todayDateComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:[NSDate date]];
+            dateComponents = todayDateComponents;
+        } else {
+            for (NSString *component in dayOfWeekMonthDayComponents) {
+                // check if the month and day of the component is found in the column and set the components respectively.
+                if ([self.class.__kifDateFormatter.monthSymbols containsObject:component]) {
+                    dateComponents.month = [self.class.__kifDateFormatter.monthSymbols indexOfObject:component] + 1;
+                } else if ([self.class.__kifDateFormatter.shortMonthSymbols containsObject:component]) {
+                    dateComponents.month = [self.class.__kifDateFormatter.shortMonthSymbols indexOfObject:component] + 1;
+                } else if ([component integerValue] != 0) { // days can't be zero and if a string is not a number it will default to zero.
+                    dateComponents.day = [component integerValue];
+                }
+            }
+        }
```

Should there be an else with a throw here for unsupported mode?

---

If it's anything else it means it's a countdown timer which shouldn't do anything with date components